### PR TITLE
tests: Fix orchestrator unstable test

### PR DESCRIPTION
**Pull request by michaljurecko**
Jira: ?

**Changes:**
- Fixed unstable test.
- Orchestrator is using restartable streams - removed code duplicity.

-----------

### CODE REVIEW

File: internal/pkg/service/common/task/orchestrator/orchestrator.go [hunk 1]
```diff
@@ -27,95 +25,47 @@ type orchestrator[T any] struct {
...
-		defer o.logger.Info(o.node.ctx, "stopped")
-		b := newRetryBackoff()
+		ctx, span := o.node.tracer.Start(o.node.ctx, spanName, trace.WithAttributes(attribute.String("resource.name", o.config.Name)))
+		stream := o.config.Source.WatchPrefix.GetAllAndWatch(ctx, o.node.client, o.config.Source.WatchEtcdOps...)
+		err := <-stream.SetupConsumer(o.logger.WithComponent("watch.consumer")).
+			WithOnClose(func(err error) {
+				span.End(&err)
+			}).
+			WithForEach(func(events []etcdop.WatchEventT[T], header *etcdop.Header, _ bool) {
+				for _, event := range events {
```
Comment (michaljurecko) [inline internal/pkg/service/common/task/orchestrator/orchestrator.go:40]: Orchestrator was created before the possibility to restart watch streams was added to the `etcdop`.
The code is now easier and there is no duplicity.

**Comment by michaljurecko** [comment]
> I'm guessing it would be best practice to always use context.WithCancelCause from now on, ideally refactor existing use cases and eventually disallow context.WithCancel in forbidigo?

:man_shrugging: : It's relatively simple to replace `cancel()` -> `cancel(context.Cancelled)`, in all places.

File: internal/pkg/service/common/etcdop/watch.go [hunk 2]
```diff
@@ -132,11 +133,11 @@ func (s RestartableWatchStream) Restart() {
 //   - The operation can be cancelled using the context.
 func (v Prefix) GetAllAndWatch(ctx context.Context, client *etcd.Client, opts ...etcd.OpOption) *RestartableWatchStream {
 	return wrapStreamWithRestart(ctx, func(ctx context.Context) *WatchStream {
-		ctx, cancel := context.WithCancel(ctx)
+		ctx, cancel := context.WithCancelCause(ctx)
```
[the remark this answers is not in the document]
Comment (Matovidlo) [reply internal/pkg/service/common/etcdop/watch.go:136]: Create task we can discuss it. I think in some cases you maybe want just plain cancel without cause?

**Review by Matovidlo — APPROVED** [review]
LGTM

**Pull request merged**